unit-tests/src/test/scala/java/io/DataOutputStreamTest.scala

```diff
@@ -0,0 +1,259 @@
+package java.io
+
+import org.junit.Test
+import org.junit.Assert._
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+
+import scalanative.junit.utils.AssertThrows._
+
+import scalanative.unsafe.sizeof
+
+class DataOutputStreamTest {
```

Comment:
Consider reusing the test suite that already exists in Scala.js, rather than inventing a new one:
https://github.com/scala-js/scala-js/blob/92c64df83377d47c55d3de0ff78d10da3ba06bcc/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/DataOutputStreamTest.scala

Reply:
Good suggestion!  I had forgotten to look at Scala.js prior art.

This `DataOutputStream` PR passed all but one Scala.js test
as is.  That builds my confidence and, I hope, the confidence of reviewers.

The failing test expected an Exception for long output strings. I ported
the Scala.js code for the failing `writeUTF()` method.

All `DataOutputStream` tests ported from Scala.js now pass.